### PR TITLE
Feature/178 enable osmnetworkwrapper creation without providing a file

### PR DESCRIPTION
**Pull request by Carsopre**

### CODE REVIEW

File: tests/network/network_wrappers/test_osm_network_wrapper.py [hunk 1]
```diff
@@ -277,3 +274,21 @@ def test_snap_nodes_to_nodes(self, _valid_graph_fixture: MultiDiGraph):
...
         assert isinstance(_result_graph, MultiDiGraph)
+
+    @slow_test
+    def test_given_valid_base_geometry_with_polygon(
+        self, _valid_network_polygon_fixture: BaseGeometry
+    ):
+        # 1. Define test data.
+        _network_config_data = self._get_dummy_network_config_data()
+        _network_config_data.network.network_type = NetworkTypeEnum.DRIVE.config_value
+        _network_config_data.network.road_types = ""
```
Comment (ArdtK) [inline tests/network/network_wrappers/test_osm_network_wrapper.py:285]: Why need this line?

**Review by ArdtK — APPROVED** [review]
Few suggestions/questions

File: ra2ce/network/network_wrappers/osm_network_wrapper/osm_network_wrapper.py [hunk 2]
```diff
@@ -121,25 +141,25 @@ def _export_linking_tables(self, linking_tables: tuple[Any]) -> None:
...
-                "No polygon_file file found at {}.".format(self.polygon_path)
-            )
-
-        _normalized_polygon = nut.get_normalized_geojson_polygon(self.polygon_path)
+        if not isinstance(polygon_path, Path):
+            logging.warning("No valid value provided for polygon file.")
+            return None
+        elif not polygon_path.is_file():
+            logging.error("No polygon_file file found at {}.".format(polygon_path))
+            return None
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply ra2ce/network/network_wrappers/osm_network_wrapper/osm_network_wrapper.py:160]: Unfortunately it is needed as `polygon_path` (`config_data.network.polygon` during the `init`) is no longer required, so if we were to trigger this method during `__init__` we would directly get an exception.

File: tests/network/network_wrappers/test_osm_network_wrapper.py (hunk 1 above)
Comment (Carsopre) [reply tests/network/network_wrappers/test_osm_network_wrapper.py:285]: It is just about providing a working case.

File: ra2ce/network/network_wrappers/osm_network_wrapper/osm_network_wrapper.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArdtK) [reply ra2ce/network/network_wrappers/osm_network_wrapper/osm_network_wrapper.py:162]: Clear, thanks

**Review by ArdtK — APPROVED** [review]
1 small comment

**Pull request merged**